File: Forum.UI/src/app/models/topico.ts
import { Comentario } from "./comentario";
import { Curtida } from "./curtida";

export interface Topico {
    id: number;
    dataHoraPostagem?: Date;
    usuarioId: number;
    conteudo: string;
    curtidas: Curtida[];
    comentarios: Comentario[];

    novoComentario: string;
    usuarioNome: string;
    expandido: boolean;
    caracteresRestantes: number;
    topicoCurtido: boolean;
    areaComentarioExpandido: boolean;
}

File: ForumMinimalAPI/Models/Topico.cs
namespace ForumMinimalAPI.Models
{
    public class Topico
    {
        public int Id { get; set; }
        public DateTime DataHoraPostagem { get; set; }
        public int UsuarioId { get; set; }
        public required string Conteudo { get; set; }
        public List<Curtida>? Curtidas {  get; set; }
        public List<Comentario>? Comentarios { get; set; }
    }
}